distro/config.go

```diff
@@ -62,6 +86,81 @@ func (c config) Validate() error {
 	return nil
 }
 
+type nonePropagatorType struct{ propagation.TextMapPropagator }
+
+// nonePropagator signals the disablement of setting a global
+// TextMapPropagator.
+var nonePropagator = nonePropagatorType{}
+
+// propagators maps environment variable values to TextMapPropagator creation
+// functions.
+var propagators = map[string]func() propagation.TextMapPropagator{
+	// W3C Trace Context.
+	"tracecontext": func() propagation.TextMapPropagator {
+		return propagation.TraceContext{}
+	},
+	// W3C Baggage
+	"baggage": func() propagation.TextMapPropagator {
+		return propagation.Baggage{}
+	},
+	// B3 Single
+	"b3": func() propagation.TextMapPropagator {
+		return b3.New(b3.WithInjectEncoding(b3.B3SingleHeader))
+	},
+	// B3 Multi
+	"b3multi": func() propagation.TextMapPropagator {
+		return b3.New(b3.WithInjectEncoding(b3.B3MultipleHeader))
+	},
+	// Jaeger
+	"jaeger": func() propagation.TextMapPropagator {
+		return jaeger.Jaeger{}
+	},
+	// AWS X-Ray.
+	"xray": func() propagation.TextMapPropagator {
+		return xray.Propagator{}
+	},
+	// OpenTracing Trace
+	"ottrace": func() propagation.TextMapPropagator {
+		return ot.OT{}
+	},
+	// None, explicitly do not set a global propagator.
+	"none": func() propagation.TextMapPropagator {
+		return nonePropagator
+	},
+}
+
+func loadPropagator(name string) propagation.TextMapPropagator {
+	var props []propagation.TextMapPropagator
+	for _, part := range strings.Split(name, ",") {
+		factory, ok := propagators[part]
+		if !ok {
+			// Skip invalid data.
+			continue
```

Comment:
can we return an error? 

If something goes wrong during initialization then we are returning an error: https://github.com/signalfx/splunk-otel-go/blob/main/distro/otel.go

Reply:
What would we do with the error? Do we want to fail if this situation is encountered, or ultimately just move on?

Reply:
Personally I would suggest failing as this would be an unsupported configuration

Reply:
We need to add logging support in the future to be compliant with other areas of the GDI specification. This seems likes it might warrant a log message. What do you think? I can add a comment to fix in the future if you think so.

Reply:
The error-handling my preference is:
- fail-fast and return error when the issue is connected with bad initialization
- log when the issue is "at runtime" - however, these issues are often transient and can make "a mess" in logs so this can be revisited later (I remember that the OTel collector has some good guidelines here)

It is compliant with https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#basic-error-handling-principles

> The API or SDK may fail fast and cause the application to fail on initialization, e.g. because of a bad user config or environment, but MUST NOT cause the application to fail later at run time, e.g. due to dynamic config settings received from the Collector.

Some functions in OTel SDK are already returning errors when the provided data is wrong e.g. [here](https://github.com/signalfx/splunk-otel-go/blob/9a017c8ef6c1553cce1a6dd7abfd5e093b5df6e0/distro/otel.go#L62)

Whatever approach/guidelines regarding error-handling you (or we) prefer I suggest documenting it in `CONTRIBUTING.md` (of course in a separate PR). Here is a simple example we have in OTel .NET Auto-Instrumentation: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/DESIGN.md#error-handling It will facilitate future discussions, reviews, development etc

Reply:
I am OK leaving a TODO/FIXME comment if you want to think about it more or address it later.